zerver/lib/validator.py

```diff
@@ -706,6 +704,8 @@ def items(self) -> Iterator[Tuple[str, "WildValue"]]:
 
 
 def wrap_wild_value(var_name: str, value: object) -> WildValue:
+    if type(value) in [WildValue, WildValueDict, WildValueList]:
```

Comment:
@andersk I encountered a situation where I need to initialise a WildValueList containing WildValueDict in one of the view.py code (zerver/webhooks/gitlab/view.py:L188-L190).

```
bar = wrap_wild_value("bar", {"baz": 1})
foo = wrap_wild_value("foo", [bar])
```

When I later try to access `foo[0]["baz"]`, zerver/lib/validator.py throws a ValidationError that complains `foo[0]` isn't a dictionary. I realised this is due to when https://github.com/zulip/zulip/blob/main/zerver/lib/validator.py#L668 is invoked when `__getitem__` is called, https://github.com/zulip/zulip/blob/main/zerver/lib/validator.py#L709-L712 doesn't check if the `value` is already a WildValue/WildValueDict/WildValueList, so it proceeded to create a WildValue with WildValueDict as its value - a nested WildValue. Then when we call `foo[0]["baz"]` it will fail because `foo[0]` is a WildValue and not a WildValueDict. Hence this fix.

However, mypy is complaining that "zerver/lib/validator.py:708: error: Incompatible return value type (got "object", expected "WildValue")  [return-value]". I can't think of a way to fix this complaint, do you have any hint?